### DISCUSSION TITLE
Tap event after pinch gesture

### DISCUSSION THREAD

**Discussion by Deodes** [software-mansion/react-native-gesture-handler#1700] [Q&A]
Helll friends, faced a problem with `<PinchGestureHandler>` which wraps big `<Pressable>` area. After finishing pinching event it's trigger `<Pressable>` handler (because of finger/fingers staying on it after pinch gesture) and i cannot prevent it. And If it is not difficult for you, please explain how we can prevent this. And how the events triggered in the react native. Because on the web there is such a thing as capturing and bubbling..

**Comment by JasonCareter** [comment]
well, you can create a value with useRef, like below
```typescript
const isPinched = useRef(false);
```
on PinchGestureHandler's onEnd callback , you can set it to true
```typescript
isPinched.current = true;
```

inside your onPress function , you can tell if user pinched before, not pinched, trigger your function
```typescript
const onPress = () => {
   if (isPinched.current) {
         isPinched.current = false;
         return;
   }
   // no pinch , just tap
   // run your onPress code here
}
```

**Comment by JasonCareter** [comment]
by the way , in the latest RNGH , you can use Gesture.Race to achieve the result. Only one gesture will activate one time. If you pinched , the tap gesture won't activate . If you tap (allow multiple fingers) , the pinch gesture won't activate.
